syntax = "proto3";


package poster;

service Poster {
    rpc PostMessages(PostMessagesRequest) returns (PostMessagesResponse);
}

message PostMessageRequest {
    string message = 1;
    MastadonAccount mastadon_account = 2;
    TwitterAccount twitter_account = 3;
}
message PostMessagesRequest {
    repeated PostMessageRequest post_message_requests = 1;
    repeated PostMessageWithImageRequest post_message_with_image_requests = 2;
    repeated PostMessageWithVideoRequest post_message_with_video_requests = 3;
}

message PostMessageWithImageRequest {
    string message = 1;
    MastadonAccount mastadon_account = 2;
    TwitterAccount twitter_account = 3;
    string image = 4;
}

message PostMessageWithVideoRequest {
    string message = 1;
    MastadonAccount mastadon_account = 2;
    TwitterAccount twitter_account = 3;
    string video = 4;
}

message PostMessageResponse {
    string response = 1;
}

message PostMessagesResponse {
    repeated PostMessageResponse post_messages_responses = 1;
}

message TwitterAccount {
    string consumer_key = 1;
    string consumer_secret = 2;
    string access_token = 3;
    string access_secret = 4;
}

message MastadonAccount {
    string server = 1;
    string client_id = 2;
    string client_secret = 3;
    string access_token = 4;
    string email = 5;
    string password = 6;
}